server/src/main/java/org/folio/reservoir/server/UploadService.java

```diff
@@ -36,6 +40,16 @@ private Future<Void> uploadPayloadStream(ReadStream<JsonObject> upload,
     AtomicInteger errors = new AtomicInteger();
     upload.exceptionHandler(promise::tryFail);
     upload.handler(r -> {
+      if (jsonPath != null) {
+        JsonObject payload = r.getJsonObject("payload");
+        Collection<String> strings = jsonPath.executeAsCollection(null, payload);
+        Iterator<String> iterator = strings.iterator();
+        if (iterator.hasNext()) {
+          r.put("localId", iterator.next().trim());
+        } else {
+          r.remove("localId");
+        }
+      }
```

Comment:
The price we pay for this is that localIdPath calculation is done one at a time. It used to be in the write method which takes place in the background.

Reply:
Because the json path is not executeblocking. If it was real javascript module it would make a difference.

Reply:
Hold on, I'll change that.

Reply:
I think the whole default local path is not so smart given that it may come even in marc from beyond 001.

Reply:
We want to be able to throw sophisticated algorithm for localid. Just like mergekeys. That's possible to execute concurrently.

Reply:
It's OK to have a default path when no localIdPath is specified, 001 is very often the right thing and reading it in code will be fastest. But we can't be falling back to it -- we don't want "mixed" IDs.